pkg/group/mysql.go

```diff
@@ -59,27 +61,46 @@ type MySQLBackend struct {
 	throttler   *throttle.Throttler
 }
 
-const maxConnections = 3
-const electionExpireSeconds = 5
+const (
+	maxConnections        = 3
+	electionExpireSeconds = 5
+
+	electionInterval = time.Second
+	healthInterval   = 2 * electionInterval
+	stateInterval    = 10 * time.Second
 
-const electionInterval = time.Second
-const healthInterval = 2 * electionInterval
-const stateInterval = 10 * time.Second
+	connectionTimeout = 500 * time.Millisecond
+)
 
 func NewMySQLBackend(throttler *throttle.Throttler) (*MySQLBackend, error) {
-	if config.Settings().BackendMySQLHost == "" {
+	settings := config.Settings()
+	if settings.BackendMySQLHost == "" {
 		return nil, nil
 	}
-	dbUri := fmt.Sprintf("%s:%s@tcp(%s:%d)/%s?interpolateParams=true&charset=utf8mb4,utf8,latin1&timeout=500ms",
-		config.Settings().BackendMySQLUser, config.Settings().BackendMySQLPassword, config.Settings().BackendMySQLHost, config.Settings().BackendMySQLPort, config.Settings().BackendMySQLSchema,
-	)
-	db, _, err := sqlutils.GetDB(dbUri)
+	uri, err := mysql.MakeUri(
+		settings.BackendMySQLHost,
+		settings.BackendMySQLPort,
+		settings.BackendMySQLSchema,
+		settings.BackendMySQLUser,
+		settings.BackendMySQLPassword,
+		settings.BackendMySQLTlsCaCertPath,
+		settings.BackendMySQLTlsClientCertPath,
+		settings.BackendMySQLTlsClientKeyPath,
+		settings.BackendMySQLTlsSkipVerify,
+		connectionTimeout)
+
 	if err != nil {
 		return nil, err
 	}
+
+	db, _, err := sqlutils.GetDB(uri)
+	if err != nil {
+		return nil, err
+	}
+
 	db.SetMaxOpenConns(maxConnections)
 	db.SetMaxIdleConns(maxConnections)
-	log.Debugf("created db at: %s", dbUri)
+	log.Debugf("created db at: %s", uri)
```

Comment:
URI contains a plaintext password, which should be scrubbed. This issue existed from before your PR, but since you've modified the handling of URI, perhaps it's a good opportunity to address the issue.

Reply:
Right on, I will address this in a separate commit! Good call!